ConstraintLayoutExamples/motionlayout/src/main/java/com/google/androidstudio/motionlayoutexample/histogramdemo/HistogramWidget.kt

```diff
@@ -0,0 +1,245 @@
+package com.google.androidstudio.motionlayoutexample.histogramdemo
+
+import android.content.Context
+import android.graphics.drawable.ColorDrawable
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.constraintlayout.motion.widget.MotionLayout
+import androidx.constraintlayout.motion.widget.MotionScene
+import androidx.constraintlayout.motion.widget.TransitionBuilder
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.content.ContextCompat
+import com.google.androidstudio.motionlayoutexample.R
+import java.util.*
+
+class HistogramWidget : MotionLayout {
```

Comment:
It looks not natural that a class named *Widget extends MotionLayout because this doesn't look a is-a relationship.

It's up to you, but how about splitting the class and rename class as *Layout and keep only the required methods for the layout class?

Reply:
Thank you for the comment. Hmm interesting point. I can see a split for sure, maybe a controller so I can achieve better MVC.

For now though, I think there're bit more benefits in leaving it as is. I'll leave it for now. :) Let me know if you want to chat more about this! I really appreciate the comment.